src/parser/jobs/mnk/RiddleOfFire.js

```diff
@@ -0,0 +1,160 @@
+import React, {Fragment} from 'react'
+import {Accordion} from 'semantic-ui-react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import Rotation from 'components/ui/Rotation'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {matchClosestHigher} from 'utilities'
+
+const ROF_DURATION = STATUSES.RIDDLE_OF_FIRE.duration * 1000
+
+const POSSIBLE_GCDS = 9
+
+export default class RiddleOfFire extends Module {
+	static handle = 'riddleoffire'
+	static dependencies = [
+		'suggestions',
+	]
+
+	static title = 'Riddle of Fire'
+
+	_active = false
+	_history = []
+	_riddle = {}
+	_rushing = false
+
+	_missedGcds = 0
+	_missedTks = 0
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.RIDDLE_OF_FIRE.id}, this._onDrop)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const actionId = event.ability.guid
+
+		if (actionId === ACTIONS.RIDDLE_OF_FIRE.id) {
+			this._active = true
```

Comment:
Same comment here as above - `hasStatus()` is a safer bet than a manual `_active` flag.